torch/testing/_internal/distributed/distributed_test.py

```diff
@@ -7080,7 +7080,7 @@ def test_ddp_inference(self):
                 device_ids=[rank]
             )
             inp = torch.randn(10, 2, device=rank)
-            inp_syncbn = torch.randn(10, 2, 4, 4, device=rank)
+            inp_syncbn = torch.randn(4, 4, 10, 2, device=rank)
```

Comment:
could this be related to the recent test_ddp_inference instability?

Reply:
This is fishy, `SyncBatchNorm` is constructed with num_channels=2, which means that input channels should be equal to 2, like it used to be before? cc @jjsjann123. Also, if error is complaining about matmul, it's coming from `model`, which is `Net()` (it has linears), not from `syncbn_model`, which doesn't have matmuls at all. `2,10` size is also in Net, not in syncbm_model`

Reply:
yes I felt exactly the same here. but yeah i think line 7077 is the culprit. will address along with a test to catch the RTE

Reply:
Hmm, not sure why this test was passing earlier. Will take a look.

Reply:
This was passing because it compared fc model (twice). The second time was with incorrect size inputs that were not erroring out before this PR, so probably the test was flaky. SyncBatchNorm was never run, and now when Rong enabled SyncBatchNorm, the test hangs.

Reply:
this also seem interesting --> according to the original issue #61291, every time the `mm` operation is invoked the result print are different. so I guess I still down understand why comparing the fc model twice yield assertEqual true both times. but maybe I am missing something here